include/circt/Dialect/Calyx/Calyx.td

```diff
@@ -38,7 +38,14 @@ def CalyxDialect : Dialect {
 
 /// Base class for the operation in this dialect.
 class CalyxOp<string mnemonic, list<OpTrait> traits = []> :
-    Op<CalyxDialect, mnemonic, traits>;
+  Op<CalyxDialect, mnemonic, traits>;
+
+/// Base class for Calyx containers.
+class CalyxContainer<string mnemonic, list<OpTrait> traits = []> :
+  Op<CalyxDialect, mnemonic, traits> {
```

Comment:
Could this specify some default traits for all of these? It seems like they'll all want `SingleBlock` and `NoTerminator`, and perhaps `HasParent<ComponentOp>`. You could take the `traits` list passed in from above for any other traits, and concatenate it with the base traits they should all have (`SingleBlock`, `NoTerminator`, etc.)

Reply:
Done. Good call, thanks.